File: gpt-3.5-finetuned/stage2/loop_construct_gang_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:loop construct gang clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    real_t * a1 = new real_t[10 * n];
    real_t * a2 = new real_t[10 * n];

    for (int x = 0; x < 10 * n; ++x){
        a1[x] = 0;
        a2[x] = rand() / (real_t)(RAND_MAX / 10);
    }

    #pragma acc data copyin(a2[0:10*n]) copyout(a1[0:10*n])
    {
        #pragma acc parallel loop gang private(temp)
        for (int y = 0; y < 10; ++y){
            temp = 0;
            #pragma acc loop gang
            for (int x = 0; x < n; ++x){
                temp = temp + a2[y * n + x];
            }
            #pragma acc loop gang
            for (int x = 0; x < n; ++x){
                a1[y * n + x] = a1[y * n + x] + a2[y * n + x] / temp;
            }
        }
    }

    for (int y = 0; y < 10; ++y){
        temp = 0;
        for (int x = 0; x < n; ++x){
            temp = temp + a2[y * n + x];
        }
        for (int x = 0; x < n; ++x){
            if (fabs(a1[y * n + x] - a2[y * n + x] / temp) > PRECISION){
                err += 1;
            }
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-finetuned/stage2/loop_construct_gang_clause.f90
!T1:loop construct gang clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8) :: a(2000), b(2000), c(2000)        
        INTEGER :: i

        SEEDDIM(1) = 1
        SEEDDIM(2) = 1
        CALL RANDOM_SEED(PUT=SEEDDIM)
        CALL RANDOM_NUMBER(a)
        CALL RANDOM_NUMBER(b)
        c = 0
          
        !$acc data copyin(a(1:2000), b(1:2000)) copy(c(1:2000))
          !$acc parallel
            !$acc loop gang
            DO i = 1, 100
              c(i) = a(i) + b(i)
            END DO
          !$acc end parallel
        !$acc end data

        DO i = 1, 100
          IF (abs(c(i) - (a(i) + b(i))) .gt. 1e-15) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM